scalafix-core/shared/src/main/scala/scalafix/rule/Rule.scala

```diff
@@ -42,7 +42,7 @@ import metaconfig.Configured
   *   Example good name: NoVars, ExplicitUnit.
   *   Example bad name: no-vars, noVars, FixVars.
   */
-abstract class Rule(ruleName: RuleName) { self =>
+abstract class Rule(ruleName: RuleName, ruleDescription: String) { self =>
```

Comment:
Can we turn this into a `def description: String = ""` and `override def description = ...` in the rule implementations? This change is binary and source breaking. I suspect that for v0.6 we will likely remove the `ruleName` constructor parameter into `def name: RuleName` and the semantic `index` into a `def withIndex(index: SemanticdbIndex): Rule` builder method. Custom rules are classloaded so making the constructors 0-arg  will greatly simplify things.

Reply:
I take that we want to bring this back to a single constructor parameter, and avoid to make it parameterless. 
Could you provide some references to make clearer to myself what this mean: "Custom rules are classloaded so making the constructors 0-arg will greatly simplify things."?